src/Blazor.Gitter.Core/Components/Pages/Index.razor.cs

```diff
@@ -13,76 +13,39 @@ public class IndexModel : ComponentBase
         [Inject] internal IAppState State { get; set; }
 
         internal string ErrorMessage;
+        string apiKey;
 
         internal string ApiKey
         {
             get => State.GetApiKey();
-            set => State.SetApiKey(value);
+            set => apiKey = value;
+        }
+
+        protected override void OnInit()
+        {
+            base.OnInit();
+            State.GotChatUser += () => StateHasChanged();
         }
 
         internal async Task SignIn(bool remember)
         {
             Console.WriteLine($"Index: SignIn {(remember ? "remember" : "")}");
 
-            if (!State.HasApiKey)
+            if (!State.HasApiKey && string.IsNullOrWhiteSpace(apiKey))
             {
-                ErrorMessage = "Did you notice that box? It says Enter your own Gitter API Key!";
+                ErrorMessage = "Please Enter your own Gitter API Key!";
                 return;
             }
 
+            State.SetApiKey(apiKey);
+
             if (remember)
             {
                 await State.SaveApiKey();
-                await AttemptLogin();
```

Comment:
We need the MainLayout (or something else that is always loaded) to handle login, so removed from here - this page is just for accepting an Api key and storing it in the AppState - and displaying user information.